doc/developer-guide.md

```diff
@@ -0,0 +1,260 @@
+# ddf-validation Developer's guide 
+
+## Registry
+
+First of all new rule should be added to the [rules registry](../lib/ddf-rules/registry.js).
+
+A new rule should be added next way:
+
+```javascript
+exports.MY_NEW_RULE = Symbol.for('MY_NEW_RULE');
+```
+
+Note: `Symbol.for` syntax is mandatory.
+
+Also, this rule should be added to descriptions:
+
+```javascript
+exports.descriptions = {
+  /// ...
+  [exports.MY_NEW_RULE]: 'Description for this rule',
+  /// ...
+};
+```
+
+## Rules applying explanation
+
+There are next types of rules:
+
+* General rules
+* Index rules
+* Concept rules
+* Entity rules
+* Data points rules
+
+### General rules
+
+This kind of rules is about situation when program doesn't know about Concepts, Entities and Data points.
+In fact, this kind of rule is not DDF based. For example, rule `Folder is not DDF`.
+
+### Index rules
+
+This kind of rules is exclusively about `ddf--index` file. For example, `Index is not found`
+or `File in index doesn't exists`.
+
+`Index rules` are placed [here](../lib/ddf-rules/index-rules.js).
+
+### Concept rules
+
+This kind of rules is exclusively about Concepts. For example, `Concept ID is not unique`.
+
+`Concept rules` are placed [here](../lib/ddf-rules/concept-rules.js).
+
+### Entity rules
+
+This kind of rules is exclusively about Entities. For example, `Entity header is not Concept`.
+
+`Entity rules` are placed [here](../lib/ddf-rules/entity-rules.js).
+
+### Data points rules
+
+This kind of rules is exclusively about Data points. For example, `Unexpected time value`.
+
+`Data points` are placed [here](../lib/ddf-rules/data-point-rules.js).
+
+## Issue object
+
+Code is placed [here](../lib/ddf-rules/issue.js).
+
+A simple way to create Issue object is:
+
+```javascript
+const registry = require('./registry');
+const Issue = require('./issue');
+// ....
+const issue = new Issue(registry.MY_NEW_RULE);
+```
+
+Constructor `Issue` has only one argument - type of issue in accordance with the registry.
+
+Next section will be based on [this document](https://github.com/jsigbiz/spec) that needed for
+methods signatures describing.
+
+Also `Issue` class has methods for additional data filling:
```

Comment:
With a help of following methods you can populate `Issue` instance with additional information regarding rule violation